### ISSUE TITLE
Move single cell and spatial library metadata handling in `Project::load_data` to their own methods

### ISSUE THREAD

**Issue by avrohomgottlieb** [AlexsLemonade/scpca-portal#645]
### Context

The beginning of `Project::load_data` deals with metadata handling (parsing, extracting, collecting, combining, etc). At first, the general `samples_metadata.csv` file is parsed, followed by the parsing and collecting of library metadata in a `<modality>.json` file (modalities being single_cell, spatial, and multiplexed).

### Problem or idea

From a modularity perspective, there is room to say that having the handling of single cell and spatial library metadata inside of `Project::load_data` contributes to code bloat within the method.

### Solution or next step

The solution is to move the handling of single cell and spatial library metadata out of `Project::load_data` and into a generic method that can handle them both.

Additionally, they both update a variety of similar data structures within their execution, which should be passed by reference and handled within the method.

The stub of the method would look as follows:
```python
def parse_library_metadata(
    self,
    modality: str,
    modality_libraries_metadata: List,
    has_modality_data: Dict,
    sample_dir: Path,
    sample_seq_units: Set,
    sample_technologies: Set,
    sample_cell_count_estimates: Optional[List] = None
) -> None:
```

**Comment by avrohomgottlieb** [comment]
Among the data structures touched within this method is an accumulator `int` variable, called `sample_cell_count_estimates`, which is currently passed by value. In order to pass this state through to the new method and get around the inability to pass a variable by reference in python, this  variable should be turned into a list which can be appended to during the new method's execution and summed up later on. The fact that this list will only hold a few values makes the performance price negligible and will contribute to cleaner code.

**Comment by davidsmejia** [comment]
Closed with #652
